src/App.tsx

```diff
@@ -1,61 +1,127 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todos } from './types/Todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const findUser = (userId: number) => {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+};
+
+const todos: Todos[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [newTodos, setNewToDos] = useState(todos);
+  const [valueTitle, setValueTitle] = useState('');
+  const [indexSelectName, setIndexSelectName] = useState(0);
+  const [checkChooseTitle, setCheckChooseTitle] = useState(false);
+  const [checkChooseUser, setCheckChooseUser] = useState(false);
+
+  const maxId = newTodos.reduce((acc, curr) => (acc.id > curr.id ? acc : curr));
+
+  const handleForm = (event: { preventDefault: () => void; }) => {
+    event.preventDefault();
+
+    if (valueTitle.length === 0) {
+      setCheckChooseTitle(true);
+    }
+
+    if (indexSelectName === 0) {
+      setCheckChooseUser(true);
+    }
+
+    if (valueTitle.length > 0 && indexSelectName > 0) {
+      const newUser = {
+        id: maxId.id + 1,
```

Comment:
u can just take `id` of the last element of array)

Reply:
id is the largest id in the array + 1 (add data-id={todo.id} attribute to each .TodoInfo). - it's not of the last element of array

Reply:
Usually, u have array sorted by ids by default, your solution is correct, I've just suggested option to make it easier